src/odemis/driver/test/npmc_test.py

```diff
@@ -235,6 +230,30 @@ def test_multimove(self):
         f2.result()
         test.assert_pos_almost_equal(orig_pos, self.dev.position.value, **COMP_ARGS)
 
+    def test_reference(self):
+        """
+        Test referencing
+        """
+
+        # Test cancellation of referencing
+        self.dev.moveAbs({'x': 0.5}).result()
+        f = self.dev.reference({'x'})
+        time.sleep(0.01)
+        f.cancel()
+        self.assertFalse(self.dev.referenced.value['x'])
+        self.assertNotEqual(self.dev.position.value['x'], 0)
+
+        # NOTE!!! If homing aborts, you MUST run referencing again. Otherwise,
```

Comment:
Well, there is probably some command that can be sent to unblock the controller without having a successful referencing... do you have any clue? If not, just put it as a "TODO" in the cancel part of the referencing code.